ui/src/app/machines/views/MachineDetails/MachineCommissioning/MachineCommissioningTable/MachineCommissioningTable.test.tsx

```diff
@@ -142,4 +155,36 @@ describe("MachineCommissioningTable", () => {
       wrapper.find("Button[data-test='action-menu-show-metrics']").exists()
     ).toEqual(false);
   });
+
+  it("sends an analytics event when clicking the 'View history' button", () => {
+    const scriptResults = [scriptResultFactory({ id: 1 })];
+    const scriptResultState = scriptResultStateFactory({
+      history: { 1: [partialScriptResultFactory()] },
+    });
+    state.nodescriptresult.items = { abc123: [1] };
+    state.scriptresult.items = scriptResults;
+    state.scriptresult = scriptResultState;
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machine/abc123", key: "testKey" }]}
+        >
+          <MachineCommissioningTable scriptResults={scriptResults} />
+        </MemoryRouter>
+      </Provider>
+    );
+
+    wrapper.find("Button.p-contextual-menu__toggle").simulate("click");
```

Comment:
Do you need to do this first? It's a shame that the tests would fail of we moved the pattern to just a button in the action column for example.

Can the test find the button even though it's hidden?

Reply:
Unfortunately we do have to do this for anything using the ContextualMenu component because it uses react portals. Basically the dropdown just straight up does not exist in the DOM until it's opened.